lib/networkx/graph.rb

```diff
@@ -23,14 +31,37 @@ def add_edges(edges)
       end
     end
 
-    def add_node(node)
-      @nodes.add(node)
+    def add_node(node, **node_attrs)
+      if !@_nodes.key?(node)
```

Comment:
In general, the `if !x` block can be written as `unless x` for more readability. But here, can't  the `!` be removed by changing the order of checking? Something like the below.

```ruby
if @nodes.key?(node)
  @nodes[node].merge!(node_attrs)
else
  @adj[node] = {}
  @nodes[node] = node_attrs
end
```